credentials/fetch_paa_certs_from_dcl.py

```diff
@@ -133,7 +139,38 @@ def use_dcld(dcld, production, cmdlist):
 @optgroup.option('--paa-trust-store-path', default='paa-root-certs', type=str, metavar='PATH', help="PAA trust store path (default: paa-root-certs)")
 def main(use_main_net_dcld, use_test_net_dcld, use_main_net_http, use_test_net_http, paa_trust_store_path):
     """DCL PAA mirroring tools"""
+    fetch_paa_certs(use_main_net_dcld, use_test_net_dcld, use_main_net_http, use_test_net_http, paa_trust_store_path)
+
+
+def get_cert_from_rest(rest_node_url, subject, subject_key_id):
+    response = requests.get(
+        f"{rest_node_url}/dcl/pki/certificates/{subject}/{subject_key_id}").json()["approvedCertificates"]["certs"][0]
+    certificate = response["pemCert"].rstrip("\n")
+    subject = response["subjectAsText"]
+    return certificate, subject
+
+
+def fetch_cd_signing_certs(store_path):
+    ''' Only supports using main net http currently.'''
+    rest_node_url = PRODUCTION_NODE_URL_REST
+    os.makedirs(store_path, exist_ok=True)
+    original_dir = os.getcwd()
+    os.chdir(store_path)
 
+    cd_signer_ids = requests.get(
+        f"{rest_node_url}/dcl/pki/child-certificates/{MATTER_CERT_CA_SUBJECT}/{MATTER_CERT_CA_SUBJECT_KEY_ID}").json()['childCertificates']['certIds']
+    for signer in cd_signer_ids:
+        subject = signer['subject']
+        subject_key_id = signer['subjectKeyId']
+        certificate, subject = get_cert_from_rest(rest_node_url, subject, subject_key_id)
+
+        print(f"Downloaded CD signing cert with subject: {subject}")
+        write_cert(certificate, subject)
+
+    os.chdir(original_dir)
+
+
+def fetch_paa_certs(use_main_net_dcld, use_test_net_dcld, use_main_net_http, use_test_net_http, paa_trust_store_path):
```

Comment:
This will include CD Signing CAs in the PAA trust store, but hose are not used as PAA resource in the DCL. The DCL stores Root CAs in the PKI schema, not PAA (only the CD Signing CAs are non-PAA though).

I'd recommend hardcoding the CD Signing Cert in the script as those are not intended to change. There is no need to query them every time. And remove them from the PAA trust store.

Reply:
PAA query shouldn't include the CD certs, as far as I can tell. This is index'd off the pki/root-certificates approvedRootCertificates list. So that should be strictly PAAs, right?

For CDs...I was torn on that. The seem like a static set, but can the CSA not mint new intermediates? ie, the root should never change, do we know the set of intermediates will not?

Reply:
From what I saw, `approvedRootCertificates` gets the data from `{rest_node_url}/dcl/pki/root-certificates`

https://on.dcl.csa-iot.org/dcl/pki/root-certificates

That one includes Root CAs, not filtered by PAA neccessarily.... so yes, it includes the CD Signing Cert (see first item in response):
> {
>   "approvedRootCertificates": {
>     "certs": [
>       {
>         "subject": "MFIxDDAKBgNVBAoMA0NTQTEsMCoGA1UEAwwjTWF0dGVyIENlcnRpZmljYXRpb24gYW5kIFRlc3RpbmcgQ0ExFDASBgorBgEEAYKifAIBDARDNUEw",
>         "subjectKeyId": "97:E4:69:D0:C5:04:14:C2:6F:C7:01:F7:7E:94:77:39:09:8D:F6:A5"
>       },
>       {
>         "subject": "MFMxCzAJBgNVBAYTAlVTMRcwFQYDVQQKEw5EaWdpQ2VydCwgSW5jLjErMCkGA1UEAxMiRGlnaUNlcnQgUm9vdCBDQSBmb3IgTUFUVEVSIFBLSSBHMQ==",
>         "subjectKeyId": "32:50:45:19:33:44:59:9B:46:65:D4:59:FD:3A:15:F1:C1:16:CC:BD"
>       },
>       {
>         "subject": "MDYxIDAeBgNVBAMMF1N0cm9uZ0tleSBNYXR0ZXIgRzEgUEFBMRIwEAYDVQQKDAlTdHJvbmdLZXk=",
>         "subjectKeyId": "37:4E:29:5A:C5:51:73:0B:AF:65:D5:29:83:9B:4E:E0:9D:39:58:8A"
>       },
>       {
>         "subject": "MDMxGzAZBgNVBAMMElRQLUxpbmsgTWF0dGVyIFBBQTEUMBIGCisGAQQBgqJ8AgEMBDExODg=",
>         "subjectKeyId": "C5:85:31:41:9E:E0:3E:E4:A1:DC:F5:51:E0:34:40:DC:2B:BE:C2:0A"
>       },
>       {

Reply:
actually, I'll just test it myself and see what's up.

Reply:
Yup - CD root was in there. Removed it now. Thanks, JC.

Reply:
I had the understanding that the SDK used the CD Signing Keys hardcoded within the SDK repo: 

- https://github.com/project-chip/connectedhomeip/tree/master/credentials/development/cd-certs
- https://github.com/project-chip/connectedhomeip/blob/master/src/credentials/attestation_verifier/DefaultDeviceAttestationVerifier.cpp